### PR TITLE
Trainer Hill

### PR DESCRIPTION
**Pull request by DizzyEggg**

### CODE REVIEW

File: src/trainer_hill.c [hunk 1]
```diff
@@ -0,0 +1,1121 @@
...
+};
+
+struct TrHillFloor
+{
+    u8 unk0;
+    u8 unk1;
+    struct TrainerHillTrainer trainers[2];
+    u8 data[0x100];
+    u16 unk3A0[16];
+    u8 coords[2]; // x first 4 bits, y last 4 bites
```
Comment (Diegoisawesome) [inline src/trainer_hill.c]: These comments don't sound right (bites?)

File: src/trainer_hill.c [hunk 2]
```diff
@@ -0,0 +1,1121 @@
...
+
+void ResetTrainerHillResults(void)
+{
+    s32 i;
+
+    gSaveBlock2Ptr->frontier.field_EF9_1 = 0;
+    gSaveBlock2Ptr->frontier.field_EF9_0 = 0;
+    gSaveBlock1Ptr->trainerHill.field_3D68 = 0;
+    for (i = 0; i < 4; i++)
+        SetTimerValue(&gSaveBlock1Ptr->trainerHillTimes[i], HILL_MAX_TIME);
```
Comment (Diegoisawesome) [inline src/trainer_hill.c]: Braces around for loops

File: src/trainer_hill.c [hunk 3]
```diff
@@ -0,0 +1,1121 @@
...
+    return gFacilityClassToTrainerClass[sRoomTrainers->facilityClass[id]];
+}
+
+void GetTrainerHillTrainerName(u8 *dst, u16 trainerId)
+{
+    s32 i;
+    u8 id = trainerId - 1;
+
+    for (i = 0; i < HILL_TRAINER_NAME_LENGTH; i++)
+        dst[i] = sRoomTrainers->name[id][i];
```
Comment (Diegoisawesome) [inline src/trainer_hill.c]: Braces around for loops

File: src/trainer_hill.c [hunk 4]
```diff
@@ -0,0 +1,1121 @@
...
+
+static void sub_81D5C5C(void)
+{
+    if (!sub_81D5C18())
+        gSpecialVar_Result = 0;
+    else
+        gSpecialVar_Result = 1;
+}
+
+void nullsub_129(void)
```
Comment (Diegoisawesome) [inline src/trainer_hill.c]: Why is one nullsub static and the other one not?

File: src/trainer_hill.c [hunk 5]
```diff
@@ -0,0 +1,1121 @@
...
+    mapId = GetFloorId();
+    src = gMapHeader.mapLayout->map;
+    gBackupMapLayout.map = mapArg;
+    gBackupMapLayout.width = 31;
+    gBackupMapLayout.height = 35;
+    dst = mapArg + 224;
+    for (i = 0; i < 5; i++)
+    {
+        for (j = 0; j < 16; j++)
+            dst[j] = src[j];
```
Comment (Diegoisawesome) [inline src/trainer_hill.c]: Braces around for loops

File: src/trainer_hill.c [hunk 6]
```diff
@@ -0,0 +1,1121 @@
...
+const struct WarpEvent* sub_81D6134(u8 warpEventId)
+{
+    u8 id;
+    const struct MapHeader *header;
+
+    if (warpEventId == 1)
+        return &gMapHeader.events->warps[1];
+
+    id = sub_81D6490();
+    if ((u8)(id - 1) >= 4)
```
Comment (Diegoisawesome) [inline src/trainer_hill.c]: Probably `id == 0 || id > 4`, or something similar (these casts and subtractions in a comparison usually are)

File: src/trainer_hill.c [hunk 7]
```diff
@@ -0,0 +1,1121 @@
...
+    CopyTrainerHillTrainerText(5, gSpecialVar_LastTalked);
+    sub_80982B8();
+}
+
+static void sub_81D62CC(u16 trainerId, u8 firstMonId)
+{
+    u8 trId, level;
+    s32 i, floorId, arrId;
+
+    if ((u16)(trainerId - 1) > 1)
```
Comment (Diegoisawesome) [inline src/trainer_hill.c]: Probably `trainerId == 0 || trainerId > 2`, or something similar (these casts and subtractions in a comparison usually are)

File: src/trainer_hill.c (hunk 1 above)
Comment (DizzyEggg) [reply src/trainer_hill.c]: lmao

File: src/trainer_hill.c (hunk 4 above)
Comment (DizzyEggg) [reply src/trainer_hill.c]: Because the other one is unused.

File: src/trainer_hill.c (hunk 6 above)
Comment (DizzyEggg) [reply src/trainer_hill.c]: Good call

File: src/trainer_hill.c (hunk 7 above)
Comment (DizzyEggg) [reply src/trainer_hill.c]: Good call.

**Pull request merged**